### DISCUSSION TITLE
part-ii/2014-101/

### DISCUSSION THREAD

**Discussion by tripos-comments** [tripos-education/maths-tripos-questions#209] [Question comments]
Part II 2014 Number Theory: Paper 1, Section I, $1 F$ 

https://questions.tripos.org/part-ii/2014-101/

**Comment by walrus1729** [comment]
<details>
<summary>Solution</summary>

An odd composite integer $N$ is a pseudoprime to the base $b$ (for $(b,N)=1$) if $b^{N-1} \equiv 1 \pmod{N}$.

The next part follows from the fact that $\{b: N \text{ is a pseudoprime to the base } b\} \leq (\mathbb{Z}/N\mathbb{Z})^{*}$. We can check this using the efficient subgroup test: if $b^{N-1} \equiv 1 \pmod{N}$ and $c^{N-1} \equiv 1 \pmod{N}$, then certainly $(bc^{-1})^{N-1} \equiv 1 \pmod{N}$ also. The result then follows from Lagrange's theorem, which implies that a proper subgroup of a group can have size at most half the size of the group.

One way to do the last part is as follows. We need to show that $F_n$ divides $2^{F_n - 1} - 1$. But $2^{F_n - 1} = F_{2^n} - 1$, so we need to show that $F_n$ divides $F_{2^n} - 2$. This is immediate from the following stronger claim.

Claim: For all $n \geq 1$, $F_n = F_0 F_1 \dots F_{n-1} + 2$.

Proof of Claim: By induction on $n$, the base case $n=1$ being easy to verify. For the inductive step, we have that 

$$
\begin{align*}
F_{n+1} - 2 &= (F_n - 1)(F_n - 1) - 1 \\
&= F_n(F_n - 2) \\
&= F_n(F_0 F_1 \dots F_{n-1}) \\
&= F_0 F_1 \dots F_n \\
\end{align*}
$$

</details>
